react/getting_started_with_react/keys_in_react.md

```diff
@@ -0,0 +1,286 @@
+  { task: 'feed the cat' },
+];
+
+const iterateTodos = todos.map((todo) => (<li key={uuid()}>{todo.task}</li>));
+
+function TodoList() {
+  return <ul>{iterateTodos}</ul>;
+}
+~~~
+
+Also bad
+
+This will cause unintended behavior when modifying, or filtering array element positions.
+Considering that todos would likely be updated, changed, filtered or etc, we would NOT want to use index as our key.
+
+~~~jsx
+const todos = ['mow the yard', 'Work on Odin Projects', 'feed the cat'];
+
+const iterateTodos = todos.map((todo, index) => (<div key={index}>{todo}</div>));
+
+function TodoList() {
+  return <div>{iterateTodos}</div>;
+}
+~~~
+
+This next example should be avoided and can be considered an anti-pattern in React.
+generating keys in this way will cause the `keys` to never match up between renders, leading to all your components and DOM being recreated every time. Not only is this slow, but it will also lose any user input inside the list items. Instead, use a stable ID based on the data.
+
+
+DO NOT DO:
+~~~jsx
+const todos = ['odin', 'lunch', 'more odin', 'laundry']
+const RenderList = () => {
+  return (
+    <div>
+      {todos.map((todo) => {
+        return (
+          <div key={uuid()}>
+            <div>{todo}</div>
+          </div>
+        )
+      })}
+    </div>
+  )
+}
+~~~
+
+DO:
+~~~jsx
+const newTodo = ({task, priority}) => {
+  return {
+    task,
+    priority,
+    id: uuid()  // id in included WHEN a todo is created
+  }
+}
+
+const todos = [
+  newTodo({task: 'odin', , priority: 'high'}), // id generated on creation
+  newTodo({task: 'lunch', priority: 'low' }),
+  newTodo({task: 'more odin', priority:'high'}),
+  newTodo({task: 'laundry', priority:'medium' })
+  ];
+
+  const RenderList = () => {
+    return (
+      <div>
+      {todos.map((todo) => {
+        return (
+          <div key={todo.id}>
+            task: {todo.task}
+            priority: {todo.priority}
+          </div>
+        )
+      })}
+      </div>
+    )
+  }
+~~~
+
+### Assignment
+
+Read this article: <a href="https://www.developerway.com/posts/react-key-attribute">how to use correctly use keys with examples demonstrating performance and re-renders.</a>
+
+Short video demonstrating <a href="https://youtu.be/xlPxnc5uUPQ">Index as Key being an Anti-pattern</a>
+
+### Knowledge Check
+
+- [What are keys](#what-are-keys)
+- [What are the rules for using keys?](#rules-for-using-keys)
+- [When can we use index as a key?](#when-is-it-ok-to-use-index-as-a-key)
+- [Why does React need keys](#why-does-react-need-keys)
+- [How to use keys](#examples)
+- [How do keys impact rendering, and re-rendering](#what-does-it-matter-if-the-keys-match-or-do-not-match)
+- [Where to get keys](#what-to-use-as-a-key)
+- [Patterns to avoid when using keys](#what-to-avoid-or-watch-out-for-when-using-keys)
+
+
+
+
+### Additional Resources
+
+<a href="https://beta.reactjs.org/learn/rendering-lists#keeping-list-items-in-order-with-key
+">React Beta Docs</a>
```

Comment:
The Additional Resources section does not adhere to our [Layout Style Guide](https://github.com/TheOdinProject/curriculum/blob/main/LAYOUT_STYLE_GUIDE.md)